packages/devextreme/js/__internal/grids/new/grid_core/columns_controller/columns_controller.test.ts

```diff
@@ -157,4 +157,55 @@ describe('ColumnsController', () => {
       ]);
     });
   });
+  describe('columns inference from firstItem', () => {
+    it('should initialize columns from the data\'s first item when no columns are defined', () => {
+      const dataItem = {
+        id: 1,
+        name: 'Alice',
+        age: 30,
+      };
+
+      const { columnsController } = setup({
+        dataSource: [dataItem],
+      });
+
+      const columns = columnsController.columns.unreactive_get();
+
+      expect(columns).toHaveLength(3);
+      expect(columns.map((col) => col.dataField)).toEqual(['id', 'name', 'age']);
```

Comment:
Please replace `toEqual` with snapshot of all columns to check dataType too and maybe other column options
Or let's add tests for `dataType` only with similar approach as for dataField right now